### PR TITLE
fix: add back rowcounts and byte scanned to dbt cli and run_results.json

### PR DESCRIPTION
**Pull request by nicor88**
### Description

Fix https://github.com/dbt-athena/dbt-athena/issues/374

This approach doesn't cover cases where we do batch inserts due to partition limitations.

## TODO
- [x] Test with table materialization (hive and iceberg)
- [ ] Test with incremental append materialization (hive and iceberg)
- [x] Test with incremental merge materialization (iceberg)
- [ ] Test with incremental partition overwrite materialization (hive)
- [x] Test when partition limits are reached (iceberg/hive) with table
- [x] Test when partition limits are reached (iceberg/hive) with incremental loads

## Checklist
- [x] You followed [contributing section](https://github.com/dbt-athena/dbt-athena#contributing)
- [ ] You kept your Pull Request small and focused on a single feature or bug fix.
- [ ] You added unit testing when necessary
- [ ] You added functional testing when necessary

### CODE REVIEW

**Comment by nicor88** [comment]
@svdimchenko when you have time could you have a look at this? - somehow it works for me, I want to test intensely against specific edge cases.

File: dbt/adapters/athena/connections.py [hunk 1]
```diff
@@ -237,13 +239,28 @@ def open(cls, connection: Connection) -> Connection:
...
-            data_scanned_in_bytes=cursor.data_scanned_in_bytes,
+            data_scanned_in_bytes=data_scanned_in_bytes,
         )
 
+    @staticmethod
+    def process_query_stats(cursor: AthenaCursor) -> Tuple[int, int]:
+        if all(map(cursor.query.__contains__, ["rowcount", "data_scanned_in_bytes"])):
+            try:
+                query_split = cursor.query.lower().split("select")[-1]
+                query_stats = re.search("{(.*)}", query_split)
```
Comment (svdimchenko) [inline dbt/adapters/athena/connections.py:255]: @nicor88 the transformations logic is not obvious for me. Would you mind to add some description as docstring here ?

**Review by svdimchenko — DISMISSED** [review]

File: dbt/adapters/athena/connections.py (hunk 1 above)
Comment (nicor88) [reply dbt/adapters/athena/connections.py:255]: sure think, good point, here I'm simplying saying that we need to pick only what is contained between { and }

**Review by svdimchenko — APPROVED** [review]
awesome!

**Comment by SoumayaMauthoorMOJ** [comment]
I compared `data_scanned_in_bytes` with the "Data Scanned" from the Athena recent queries tab and the value is consistently 4.8% greater. @nicor88 @svdimchenko do you have any ideas why ?

**Comment by nicor88** [comment]
As you can see from the implementation we purely pass the result of the cursor to result. 

Few ideas:
- try to run a simple query without dbt, using PyAthena and see if you get the same difference
- try out if this 4.8% difference exists in the previous version of the adapter e.g. 1.5.1
- did you try to use boto3 athena client and see what you get for the same execution id?

**Comment by SoumayaMauthoorMOJ** [comment]
Oh silly me, it's because of binary: https://www.gbmb.org/bytes-to-mb
